_posts/2023-04-13-denoising-diffusion/denoising-diffusion.Rmd

```diff
@@ -0,0 +1,208 @@
+function, training is subject to a single -- though composite -- loss.
+One component makes sure that reconstructed samples closely resemble the
+input; the other, that the latent code confirms to pre-imposed
+constraints.
+
+Lastly, let us mention **flows** (although these tend to be used for a
+different purpose, see next section). A flow is a sequence of
+differentiable, invertible mappings from data to some "nice"
+distribution, nice meaning "something we can easily sample, or obtain a
+likelihood from". With flows, like with diffusion, learning happens
+during the forward stage. Invertibility, as well as differentiability,
+then assure that we can go back to the input distribution we started
+with.
+
+Before we dive into diffusion, we sketch -- *very* informally -- some
+aspects to consider when mentally mapping the space of generative
+models.
+
+## Generative models: If you wanted to draw a mind map ...
+
+Above, I've given rather technical characterizations of the different
+approaches: What is the overall setup, what do we optimize for ...
+Staying on the technical side, we could look at established
+categorizations such as likelihood-based vs. not-likelihood-based
+models. Likelihood-based models directly parameterize the data
+distribution; the parameters are then fitted by maximizing the
+likelihood of the data under the model. From the above-listed
+architectures, this is the case with VAEs and flows; it is not with
+GANs.
+
+But we can also take a different perspective -- that of purpose.
+Firstly, are we interested in representation learning? That is, would we
+like to condense the space of samples into a sparser one, one that
+exposes underlying features and gives hints at useful categorization? If
+so, VAEs are the classical candidates to look at.
+
+Alternatively, are we mainly interested in generation, and would like to
+synthesize samples corresponding to different levels of coarse-graining?
+Then diffusion algorithms are a good choice. It has been shown that
+
+> [...] representations learnt using different noise levels tend to
+> correspond to different scales of features: the higher the noise
+> level, the larger-scale the features that are captured.[^1]
+
+[^1]: From @dieleman2022diffusion, an excellent high-level introduction
+    to diffusion models.
+
+As a final example, what if we aren't interested in synthesis, but would
+like to assess if a given piece of data could likely be part of some
+distribution? If so, flows might be an option.[^2]
+
+[^2]: For more information on this lesser-used family of algorithms, see
+    <https://blog.evjang.com/2018/01/nf1.html>.
+
+## Zooming in: Diffusion models
+
+Just like about every deep-learning architecture, diffusion models
+constitute a heterogeneous family. Here, let us just name a few of the
+most en-vogue members. [^3]
+
+[^3]: En vogue as of this writing (the usual caveat that comes with
+    using epitheta such as this).
```

Comment:
```suggestion
    using epithets such as this).
```

Reply:
Unless you _really_ want to conjugate from the Greek singular epitheton to epitheta, I'd just pluralize epithet here. If they mean something different, it's sufficiently obscure that I had to Google it, so maybe use a different word.